trackfilter.cc

```diff
@@ -78,34 +78,42 @@ int TrackFilter::trackfilter_opt_count()
 
 qint64 TrackFilter::trackfilter_parse_time_opt(const char* arg)
 {
-  qint64 result;
+  qint64 result = 0;
 
-  QRegularExpression re("^([+-]?\\d+)([dhms])$", QRegularExpression::CaseInsensitiveOption);
+  QRegularExpression re("^([+-]?\\d+)([wdhms])(?:([+-]?\\d+)([wdhms]))?(?:([+-]?\\d+)([wdhms]))?(?:([+-]?\\d+)([wdhms]))?(?:([+-]?\\d+)([wdhms]))?$", QRegularExpression::CaseInsensitiveOption);
```

Comment:
I can't read this regex now, but can it ever match a unit into match[0] but not have a value so that the offset below will walk off the end of match? Maybe track,move=+1024   (no units)

Otherwise, lgtm. Thanx

Reply:
I don't think so.
1. match.captured(0) is the whole match.  we don't use 0.
2. we use pairs of match with indexes 1 & 2, 3 & 4, ...
3. since the first index in a pair is < the lastCapturedIndex we are guaranteed that
the second index in a pair is <= lastCapturedIndex.  So even if the re can capture matches not in pairs (ignoring 0) we won't index off the end.
4. the re is anchored by ^ and $ so we shouldn't get partial matches (with a few caveats).
5. the re is build of units of ([+-]?\\d+)([wdhms]).  The first group has to find one or more digits due to the + quantifier, and the second group has to match one of the supplied characters due to the implicit quantifier of 1.   So we either get 0 matches or 2 from each of these units.